homeassistant/components/ozw/lock.py

```diff
@@ -37,3 +75,52 @@ async def async_lock(self, **kwargs):
     async def async_unlock(self, **kwargs):
         """Unlock the lock."""
         self.values.primary.send_value(False)
+
+    @callback
+    def async_set_usercode(self, code_slot, usercode):
+        """Set the usercode to index X on the lock."""
+        lock_node = self.values.primary.node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                if len(str(usercode)) < 4:
+                    _LOGGER.error(
+                        "Invalid code provided: (%s) user code must be at least 4 digits",
+                        usercode,
+                    )
+                    break
+                value.send_value(usercode)
+                _LOGGER.debug("User code at slot %s set to: %s", code_slot, usercode)
+                break
+
+    @callback
+    def async_get_usercode(self, code_slot):
+        """Get a usercode at index X on the lock."""
+        lock_node = self.values.primary.node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                _LOGGER.info("User code at slot %s is: %s", code_slot, value.value)
```

Comment:
I've been thinking about this service… it doesn't really make sense does it? It also means that now all of a sudden our logging contains sensitive information. If a user wants this information, they should just open ozwadmin ?

Reply:
fixed in latest commit